src/SignalR/clients/ts/signalr-protocol-msgpack/src/MessagePackHubProtocol.ts

```diff
@@ -32,6 +34,16 @@ export class MessagePackHubProtocol implements IHubProtocol {
     private readonly voidResult = 2;
     private readonly nonVoidResult = 3;
 
+    /**
+     *
+     * @param messagePackOptions MessagePack options passed to msgpack5
+     */
+    constructor(private messagePackOptions?: MessagePackOptions) {
+        if (messagePackOptions) {
+            (messagePackOptions as any).compatibilityMode = false;
```

Comment:
@rynowak Suggested that we clone the properties we want rather than overwrite what we don't want changed.

Reply:
Ah ok.
Are you picturing something like this:

```
constructor(messagePackOptions?: MessagePackOptions) {
        if (messagePackOptions) {
            this.messagePackOptions = this.ensureValidOptions(messagePackOptions);
        }
}

private ensureValidOptions(options: MessagePackOptions) {
        const supportedOptions: MessagePackOptions = {};
        const unsupportedOptions: string[] = ["compatibilityMode"];

        for (const prop in options) {
            if (unsupportedOptions.indexOf(prop) === -1) {
                supportedOptions[prop] = options[prop];
            }
        }

        return supportedOptions;
}
```

Reply:
You should be able to use the spread operator here.

```typescript
if (messagePackOptions) {
    this.messagePackOptions = { ...messagePackOptions };
    (this.messagePackOptions as any).compatibilityMode = false;
}
```

Reply:
Even easier, this should work as a "one-liner"

```typescript
if (messagePackOptions) {
	this.messagePackOptions = {
		...messagePackOptions,
		compatibilityMode: false
	};
}
```

The spread operator is evaluated in-order with the rest of the assignments.

Reply:
Very nice! I like this syntax - nice and compact!
Updated.